test/ping/message_test.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class PingMessageTest < Test::Unit::TestCase
+  def test_create_message
+    fake "api/ping-api/v1/conversations/123/messages", method: :post, body: load_fixture('ping/message')
+
+    message = ShopifyAPI::Ping::Message.new(
+      dedupe_key: SecureRandom.uuid,
+      content: {
+        text: "Hello from shopify_api",
+      },
+      sender_id: 'test',
+      conversation_id: '123',
+    )
+
+    message.save
+    assert_equal "d0c7a2e6-8084-4e79-8483-e4a1352b81f7", message.id
+  end
+
+  def test_successful_delivery
+    fake "api/ping-api/v1/conversations/123/messages", method: :post, body: load_fixture('ping/message')
+
+    message = ShopifyAPI::Ping::Message.new(
+      dedupe_key: SecureRandom.uuid,
+      content: {
+        text: "Hello from shopify_api",
+      },
+      sender_id: 'test',
+      conversation_id: '123',
+    )
+
+    message.save
+
+    fake "api/ping-api/v1/conversations/123/messages/#{message.id}/delivery_confirmation",
+      method: :post,
+      body: ''
+
+    delivery_confirmation = message.successful_delivery(conversation_id: '123', message_id: message.id)
```

Comment:
This only works one way. Integration -> Ping since the integration has a reference to the message object but will not work when all you have is the message_id and conversation_id.

Reply:
That is correct, will there be use cases where we'll need this to work in the opposite direction since the endpoint is specifically for messages that have been processed by an integration?